src/environment.jl

```diff
@@ -0,0 +1,42 @@
+default_environment_algorithm() = "exact"
+
+function environment(
+  ψ::AbstractITensorNetwork,
+  vertices::Vector;
+  alg=default_environment_algorithm(),
+  kwargs...,
+)
+  return environment(Algorithm(alg), ψ, vertices; kwargs...)
+end
+
+function environment(
+  ::Algorithm"exact",
+  ψ::AbstractITensorNetwork,
+  vertices::Vector;
+  contraction_sequence_alg="optimal",
+  kwargs...,
+)
+  ψ_reduced = Vector{ITensor}(subgraph(ψ, vertices))
```

Comment:
Shouldn't this be `setdiff(vertices(ψ), verts)` (where `verts` are the vertices input into the function)? I.e. the environment would be calculated by contracting the rest of the system.

Reply:
Good spot. I was handling the logic for that in the wrong place (at the level of the `form network` and so hence why it didn't come up in testing). I fixed it now and handle the `setdiff` logic at the lowest possible level.